dataRetriever/requestHandlers/requestHandler.go

```diff
@@ -173,6 +190,14 @@ func (rrh *resolverRequestHandler) RequestRewardTransactions(destShardId uint32,
 
 // RequestMiniBlock method asks for miniblocks from the connected peers
 func (rrh *resolverRequestHandler) RequestMiniBlock(destShardID uint32, miniblockHash []byte) {
+	rrh.requestedItemsHandler.Sweep()
```

Comment:
is it really ok to sweep everything here ?

Reply:
Actually now is better than before when sweep was done on each Has and Add. But now if you pointed out this I made another improvement to have sweep done only from time to time